examples/chip-tool/commands/common/Command.cpp

```diff
@@ -307,6 +319,14 @@ const char * Command::GetAttribute(void) const
     return nullptr;
 }
 
+void Command::InitInteractionModelDelegate()
+{
+    VerifyOrReturn(mpIMDelegate == nullptr);
```

Comment:
if not null, don't we still need to `SetCommandObjet` on it? You also need to check memory allocations.

```cpp
if (mIPDelegete == nullptr) {
   mIPDelegate = chip::Platform::New<ChipToolInteractionModelDelegate>();
}
if (mIPDelegate == nullptr {
   // above allocation failed ... this is unexpected
   ChipLogError(...);
   return CHIP_ERROR_NO_MEMORY;
}
mIPDelegate->SetCommandObject(this);
```